python/lsst/cp/pipe/bfCoeffCalc.py

```diff
@@ -0,0 +1,1272 @@
+        doc="""Task to perform instrumental signature removal or load a post-ISR image; ISR consists of:
+            - assemble raw amplifier images into an exposure with image, variance and mask planes
+            - perform bias subtraction, flat fielding, etc.
+            - mask known bad pixels
+            - provide a preliminary WCS
+            """,
+    )
+    doCalcGains = pexConfig.Field(
+        dtype=bool,
+        doc="Measure the per-amplifier gains using the PTC method",
+        default=True,
+    )
+    ccdKey = pexConfig.Field(
+        dtype=str,
+        doc="The key by which to pull a detector from a dataId, e.g. \'ccd\'' or \'detector\'",
+        default='ccd',
+    )
+    maxIterRegression = pexConfig.Field(
+        dtype=int,
+        doc="Maximum number of iterations for the iterative regression fitter",
+        default=10
+    )
+    nSigmaClipGainCalc = pexConfig.Field(
+        dtype=int,
+        doc="Number of sigma to clip to during gain calculation",
+        default=5
+    )
+    nSigmaClipRegression = pexConfig.Field(
+        dtype=int,
+        doc="Number of sigma to clip to during iterative regression",
+        default=3
+    )
+    xcorrCheckRejectLevel = pexConfig.Field(
+        dtype=float,
+        doc="Sanity check level for the sum of the input cross-correlations. Arrays which "
+        "sum to greater than this are discarded before the clipped mean is calculated.",
+        default=1.0  # xxx change this back once the problem is fixed!
+        # default=0.2
+    )
+    maxIterSOR = pexConfig.Field(
+        dtype=int,
+        doc="The maximum number of iterations allowed for the successive over-relaxation method",
+        default=10000
+    )
+    eLevelSOR = pexConfig.Field(
+        dtype=float,
+        doc="The target residual error for the successive over-relaxation method",
+        default=5.0e-14
+    )
+    nSigmaClipKernelGen = pexConfig.Field(
+        dtype=float,
+        doc="Number of sigma to clip to during pixel-wise clipping when generating the kernel",
+        default=4
+    )
+    nSigmaClipXCorr = pexConfig.Field(
+        dtype=float,
+        doc="Number of sigma to clip when calculating means for the cross correlation",
+        default=5
+    )
+    maxLag = pexConfig.Field(
+        dtype=int,
+        doc="The maximum lag to use when calculating the cross correlation/kernel",
+        default=8
+    )
+    nPixBorderGainCalc = pexConfig.Field(
+        dtype=int,
+        doc="The number of border pixels to exclude when calculating the gain",
+        default=10
+    )
+    nPixBorderXCorr = pexConfig.Field(
+        dtype=int,
+        doc="The number of border pixels to exclude when calculating the cross correlation/kernel",
+        default=10
+    )
+    biasCorr = pexConfig.Field(
+        dtype=float,
+        doc="A scary, empirically determined correction-factor correcting for sigma-clipping" +
+        " a non-Gaussian distribution",
```

Comment:
I've done the first bits (the phrasing etc), but the explanation is a bit tricky. Essentially this is something that probably shouldn't be touched except by experts, but I made it a config option because having a hard-coded number in the code filled me with horror, especially as I can imagine this being different in some cases. How it's calculated should indeed be added, and probably methods for doing so added to this module, and therefore those referenced here, but given that hasn't been done yet, I'm going to have to punt on this for now.

Reply:
I think what I'll do is ticket adding that code to this, and add the ticket number to this docstring, saying that it should be update once DM-XXXXX is done. Does that sound like a reasonable solution?

Reply:
Yes